### PR TITLE
Feature/api admin roles header

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Pass down admin roles and admin roles group through headers for consumption by API backends

# Impacts
API frontend

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [N/A] Document the feature
- [YES] Add unit tests
- [N/A] Add acceptance tests (TestLink)

### CODE REVIEW

File: go/api-frontend/aaa/authorization.go [hunk 1]
```diff
@@ -124,6 +124,22 @@ func (tam *TokenAuthorizationMiddleware) BearerRequestIsAuthorized(ctx context.C
 		}
 	}
 
+	roles := make([]string, len(tokenInfo.AdminRoles()))
+	i := 0
+	for r, _ := range tokenInfo.AdminRoles() {
+		roles[i] = r
+		i++
+	}
+	r.Header.Set("X-PacketFence-Admin-Roles", strings.Join(roles, ","))
```
Comment (jrouzierinverse) [inline go/api-frontend/aaa/authorization.go:133]: The admin actions are not needed.
I just need the name of the Admin roles that are in conf/adminroles.conf

File: go/api-frontend/aaa/authentication_backend_test.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func TestMemAuthenticationBackend(t *testing.T) {
 		t.Error("User was unauthenticated although it presented valid credentials. error:", err)
 	}
 
-	if !tokenInfo.AdminRoles["SYSTEM_READ"] {
+	if !tokenInfo.AdminRoles()["SYSTEM_READ"] {
```
Comment (jrouzierinverse) [inline go/api-frontend/aaa/authentication_backend_test.go]: The name AdminRoles is confusing here since in perl they are called actions not roles.

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: go/api-frontend/aaa/authorization.go (hunk 1 above)
Comment (julsemaan) [reply go/api-frontend/aaa/authorization.go:133]: I know but might as well add them for future utilization.

Comment (jrouzierinverse) [reply go/api-frontend/aaa/authorization.go:133]: Then we will add it when we need them.
If someone has the Admin Role ALL that would send an extra 2286 bytes that is wasteful.
If we need them can always add them back.

```
perl -Ilib -Mpf::constants::admin_roles -e'print(length("X-PacketFence-Admin-Roles:" . join(",", @pf::constants::admin_roles::ADMIN_ACTIONS) . "\r\n" )  ,"\n")'
```

Comment (julsemaan) [reply go/api-frontend/aaa/authorization.go:133]: Yeah you're right for the large amount of bytes, didn't think of that, I'll strip it

**Comment by julsemaan** [comment]
Performed the adjustments

**Comment by jrouzierinverse** [comment]
```
```

**Comment by jrouzierinverse** [comment]
``

**Comment by jrouzierinverse** [comment]
Cannot compile
```
[root@dev-c7 pf]# cd ~/gospace/src/github.com/inverse-inc/packetfence/go
[root@dev-c7 go]# make
make NAME="pfhttpd" _pfhttpd
make[1]: Entering directory `/usr/local/pf/go'
cd caddy/caddy/caddy && \
go build && \
cd - && \
mv caddy/caddy/caddy/caddy pfhttpd
# github.com/inverse-inc/packetfence/go/caddy/api-aaa
../../api-aaa/api-aaa.go:167:36: cannot call non-function info.AdminRoles (type map[string]bool)
make[1]: *** [_pfhttpd] Error 2
make[1]: Leaving directory `/usr/local/pf/go'
make: *** [pfhttpd-sha] Error 2
[root@dev-c7 go]# v ../../api-aaa/api-aaa.go +167
[root@dev-c7 go]# v api-frontend/
aaa/              unifiedapierrors/
[root@dev-c7 go]# v caddy/api-aaa/api-aaa
api-aaa.go       api-aaa_test.go
[root@dev-c7 go]# v caddy/api-aaa/api-aaa.go +167
[root@dev-c7 go]# ag GetTokenInfoFromBearerRequest -l
caddy/api-aaa/api-aaa.go
api-frontend/aaa/authorization.go
[root@dev-c7 go]# v api-frontend/aaa/authorization.go
```

**Comment by julsemaan** [comment]
adjusted the code

**Pull request merged**